runners/core-java/src/main/java/org/apache/beam/runners/core/StateTable.java

```diff
@@ -30,7 +30,7 @@
  */
 public abstract class StateTable {
 
-  private final Table<StateNamespace, StateTag<?>, State> stateTable =
+  private final Table<StateNamespace, String, State> stateTable =
```

Comment:
We could use an `Equivalence` instead (it would be my preference; then the Tag is still the key, it's just we only care about the ID)